### PR TITLE
Local review: Timeout & LowPowerTimeout tests

### PR DESCRIPTION
**Pull request by fkjagodzinski**
For local review only.

### CODE REVIEW

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+    T timeout;
+    Timer timer;
+
+    timer.start();
+    sleep_manager_lock_deep_sleep();
+    timeout.attach_callback(mbed::callback(sem_callback, &sem), delay_us);
+
+    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
+    TEST_ASSERT_FALSE_MESSAGE(deep_sleep_allowed, "Deep sleep should be disallowed");
+    sleep();
```
Comment (mprse) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: Consider the following modification:
`sleep();`
to:
```
flag = true;
while(flag) sleep();
```

_flag_ should be set to false in the callback function. 
This is to be sure that seep mode is not released by the other event.

Additionally it looks like this line is redundant:
`int32_t sem_slots = sem.wait(osWaitForever);`

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    /*
+     * We use here the low power timer instead of microsecond timer for start and
+     * end because the microseconds timer might be disabled during deepsleep.
+     */
+    timer.start();
+    timeout.attach_callback(mbed::callback(sem_callback, &sem), delay_us);
+    /* Make sure deepsleep is allowed, to go to deepsleep */
+    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
+    TEST_ASSERT_TRUE_MESSAGE(deep_sleep_allowed, "Deep sleep should be allowed");
+    sleep();
```
Comment (mprse) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: Please see above comment.

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+
+    int32_t sem_slots = sem.wait(osWaitForever);
```
Comment (mprse) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: If this line is needed, then it should be moved outside timer start-stop section and wait value should be set to 0.

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+    timer.start();
+    sleep_manager_lock_deep_sleep();
+    timeout.attach_callback(mbed::callback(sem_callback, &sem), delay_us);
+
+    bool deep_sleep_allowed = sleep_manager_can_deep_sleep();
+    TEST_ASSERT_FALSE_MESSAGE(deep_sleep_allowed, "Deep sleep should be disallowed");
+    sleep();
+
+    int32_t sem_slots = sem.wait(osWaitForever);
```
Comment (mprse) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: If this line is needed, then it should be moved outside timer start-stop section and wait value should be set to 0.

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_TIMEOUT_TESTS_H
+#define MBED_TIMEOUT_TESTS_H
+
+#include "mbed.h"
+#include "unity/unity.h"
+#include "rtos/rtos.h"
```
Comment (maciejbocianski) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: "rtos/rtos.h" is included inside "mbed.h"

File: TESTS/mbed_drivers/timeout/timeout_tests.h [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+#define NUM_TIMEOUTS 16
+#define DRIFT_TEST_PERIOD_US 10000
+
+const float TEST_DELAY_S = 0.01;
+const uint32_t TEST_DELAY_MS = 1000.0F * TEST_DELAY_S;
+const us_timestamp_t TEST_DELAY_US = 1000000.0F * TEST_DELAY_S;
+
+/* Timeouts are quite arbitrary due to large number of boards with varying level of accuracy */
+#define LONG_DELTA_US (100000)
+#define SHORT_DELTA_US (600)
```
Comment (maciejbocianski) [inline TESTS/mbed_drivers/timeout/timeout_tests.h]: not enough for nrf51_dk

File: TESTS/mbed_drivers/timeout/timeout_tests.h (hunk 1 above)
Comment (fkjagodzinski) [reply TESTS/mbed_drivers/timeout/timeout_tests.h]: Changed as suggested.

File: TESTS/mbed_drivers/timeout/timeout_tests.h (hunk 2 above)
Comment (fkjagodzinski) [reply TESTS/mbed_drivers/timeout/timeout_tests.h]: Fixed as suggested.

File: TESTS/mbed_drivers/timeout/timeout_tests.h (hunk 5 above)
Comment (fkjagodzinski) [reply TESTS/mbed_drivers/timeout/timeout_tests.h]: Fixed.

File: TESTS/mbed_drivers/timeout/timeout_tests.h (hunk 6 above)
Comment (fkjagodzinski) [reply TESTS/mbed_drivers/timeout/timeout_tests.h]: Increased `SHORT_DELTA_US` to 1200 us and updated test case delay values.

**Pull request closed without merging**